src/plugins/vis_type_timeseries/public/application/components/data_format_picker.tsx

```diff
@@ -0,0 +1,304 @@
+    },
+    [changeModelFormatter, durationParams]
+  );
+
+  const handleDurationChange = useCallback(
+    (optionName: 'from' | 'to') => {
+      return ([{ value }]: Array<EuiComboBoxOptionOption<string>>) =>
+        handleDurationParamsChange(optionName, value!);
+    },
+    [handleDurationParamsChange]
+  );
+
+  const handleDecimalsChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) =>
+      handleDurationParamsChange('decimals', event.target.value),
+    [handleDurationParamsChange]
+  );
+
+  const htmlId = htmlIdGenerator();
+
+  let duration;
+  if (selectedFormatter === DATA_FORMATTERS.DURATION) {
+    const { from, to, decimals = DEFAULT_OUTPUT_PRECISION } = durationParams;
+    const selectedFrom = durationInputOptions.find(({ value }) => value === from);
+    const selectedTo = durationOutputOptions.find(({ value }) => value === to);
+
+    duration = (
+      <>
+        <EuiFlexItem grow={false}>
+          <EuiFormRow
+            id={htmlId('from')}
+            label={i18n.translate('visTypeTimeseries.dataFormatPicker.fromLabel', {
+              defaultMessage: 'From',
+            })}
+          >
+            <EuiComboBox
+              isClearable={false}
+              options={durationInputOptions}
+              selectedOptions={selectedFrom ? [selectedFrom] : []}
+              onChange={handleDurationChange('from')}
+              singleSelection={{ asPlainText: true }}
+              data-test-subj="dataFormatPickerDurationFrom"
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiFormRow
+            id={htmlId('to')}
+            label={i18n.translate('visTypeTimeseries.dataFormatPicker.toLabel', {
+              defaultMessage: 'To',
+            })}
+          >
+            <EuiComboBox
+              isClearable={false}
+              options={durationOutputOptions}
+              selectedOptions={selectedTo ? [selectedTo] : []}
+              onChange={handleDurationChange('to')}
+              singleSelection={{ asPlainText: true }}
+              data-test-subj="dataFormatPickerDurationTo"
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+
+        {selectedTo?.value !== 'humanize' && (
+          <EuiFlexItem grow={false}>
+            <EuiFormRow
+              id={htmlId('decimal')}
+              label={i18n.translate('visTypeTimeseries.dataFormatPicker.decimalPlacesLabel', {
+                defaultMessage: 'Decimal places',
+              })}
+            >
+              <EuiFieldText
+                defaultValue={decimals}
+                placeholder={DEFAULT_OUTPUT_PRECISION}
+                onChange={handleDecimalsChange}
+                data-test-subj="dataFormatPickerDurationDecimal"
+              />
+            </EuiFormRow>
+          </EuiFlexItem>
+        )}
+      </>
+    );
+  }
+
+  let custom;
+  if (selectedFormatter === DATA_FORMATTERS.CUSTOM) {
+    custom = (
```

Comment:
Have two simple suggestions: 
1. For non-numeric values, hide all instead of disabling. I know we already disscussed it with you, sorry, but during testing I found that it will be better. 
2. Replace `Format string` component to the following view. See placeholder, append/prepend texts. It was copied from index-pattern-management page.

![image](https://user-images.githubusercontent.com/20072247/129369816-bff942b3-5475-4975-9305-9b1b836f2e63.png)

This simple changes should make that functionality a little bit more clear for our users 

@stratoula any ideas?

Reply:
I agree with both!